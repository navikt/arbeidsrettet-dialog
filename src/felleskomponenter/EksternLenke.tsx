import React from 'react';
import Lenke from 'nav-frontend-lenker';
import EksternLenkeIkon from './EksternLenkeIkon';

const httpRegex = /^(https?):\/\/.*$/;

interface PropTypes {
    lenke: string | null;
}

export default function EksternLenke(props: PropTypes) {
    const { lenke } = props;

    if (!lenke) return null;
    const paddaLenke = lenke && lenke.match(httpRegex) ? lenke : `http://${lenke}`;
    return (
        <Lenke href={paddaLenke}>
            {lenke}
            <EksternLenkeIkon />
        </Lenke>
    );
}
