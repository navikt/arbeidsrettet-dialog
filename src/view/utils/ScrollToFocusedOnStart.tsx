import React, { ReactElement, useEffect } from 'react';
import { RouteComponentProps, RouteProps, withRouter } from 'react-router-dom';

const ScrollToFocusedOnStart = (props: RouteProps): ReactElement<RouteComponentProps<any>> => {
    useEffect(() => document.activeElement?.scrollIntoView(false), []);
    return <>{props.children}</>;
};

export default withRouter(ScrollToFocusedOnStart);
