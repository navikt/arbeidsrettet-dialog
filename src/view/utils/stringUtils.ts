export const cutStringAtLength = (text: string, maxLength: number, postfix?: string) => {
    return text.length > maxLength
        ? `${text.substring(0, postfix ? maxLength - postfix.length : maxLength)}${postfix ? postfix : ''}`
        : text;
};
