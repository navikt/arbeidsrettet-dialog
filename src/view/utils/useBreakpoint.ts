import { useEffect, useState } from 'react';

export enum Breakpoint {
    initial = 0,
    sm = 480,
    md = 768,
    lg = 1024,
    xl = 1280,
    '2xl' = 1920
}

const getBreakpoint = (): Breakpoint => {
    const { innerWidth: width } = window;
    if (Breakpoint.initial <= width && width < Breakpoint.sm) {
        return Breakpoint.initial;
    }
    if (Breakpoint.sm < width && width < Breakpoint.md) {
        return Breakpoint.sm;
    }
    if (Breakpoint.md < width && width < Breakpoint.lg) {
        return Breakpoint.md;
    }
    if (Breakpoint.lg < width && width < Breakpoint.xl) {
        return Breakpoint.lg;
    }
    if (Breakpoint.xl < width && width < Breakpoint['2xl']) {
        return Breakpoint.xl;
    }
    return Breakpoint['2xl'];
};

export const useBreakpoint = (): Breakpoint => {
    const [breakpoint, setBreakPoint] = useState(() => getBreakpoint());
    const handleResize = () => {
        setBreakPoint(getBreakpoint());
    };

    useEffect(() => {
        window.addEventListener('resize', handleResize);
        return () => window.removeEventListener('resize', handleResize);
    }, []);

    return breakpoint;
};
