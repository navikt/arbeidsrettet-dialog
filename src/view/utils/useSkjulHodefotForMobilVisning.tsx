import { useEffect } from 'react';

const selectHodefot = () => Array.from(document.querySelectorAll('.hodefot'));

function skjulElementerForMobil() {
    selectHodefot().forEach(el => {
        el.classList.add('hidden-xs');
    });
}

function visElementer() {
    selectHodefot().forEach(el => {
        el.classList.remove('hidden-xs');
    });
}

export function useSkjulHodefotForMobilVisning() {
    useEffect(() => {
        skjulElementerForMobil();
        return () => visElementer();
    }, []);
}
