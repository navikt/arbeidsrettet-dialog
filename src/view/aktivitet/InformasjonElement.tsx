import Tekstomrade from 'nav-frontend-tekstomrade';
import { Undertekst } from 'nav-frontend-typografi';
import React, { ReactNode } from 'react';

import EksternLenke from '../../felleskomponenter/EksternLenke';
import { StringOrNull } from '../../utils/Typer';
import styles from './AktivitetskortInfoBox.module.less';

interface TekstomradeProps {
    merkelapptekst: string;
    verdi?: StringOrNull;
}

export default function InformasjonElement(props: TekstomradeProps) {
    const { verdi, merkelapptekst } = props;
    if (!verdi) {
        return <InformasjonElementRaw {...props} />;
    }

    return (
        <InformasjonElementRaw merkelapptekst={merkelapptekst}>
            <Tekstomrade className={styles.overflowEllipse}>{verdi}</Tekstomrade>
        </InformasjonElementRaw>
    );
}

export function LenkeInformasjonElement(props: TekstomradeProps) {
    const { verdi, merkelapptekst } = props;
    if (!verdi) {
        return <InformasjonElementRaw {...props} />;
    }

    return (
        <InformasjonElementRaw merkelapptekst={merkelapptekst}>
            <EksternLenke lenke={verdi} />
        </InformasjonElementRaw>
    );
}

interface PropTypes {
    merkelapptekst: string;
    children?: ReactNode;
}

export function InformasjonElementRaw(props: PropTypes) {
    const { merkelapptekst, children } = props;

    if (!children) return null;

    return (
        <div className={styles.informasjonselement}>
            <Undertekst children={merkelapptekst} className={styles.merkelapp} />
            {children}
        </div>
    );
}
