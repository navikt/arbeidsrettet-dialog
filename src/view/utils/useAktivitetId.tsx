import queryString from 'query-string';
import { useLocation, useParams } from 'react-router';

import { DialogData } from '../../utils/Typer';
import { MaybeAktivitet, findAktivitet, useAktivitetContext } from '../AktivitetProvider';
import { useDialogContext } from '../DialogProvider';

function getFirst(maybeArray: string | (string | null)[]): string | undefined {
    if (Array.isArray(maybeArray)) {
        const first = maybeArray[0];
        return first ?? undefined;
    }
    return maybeArray;
}

export function useAktivitetId(): string | undefined {
    const location = useLocation();
    const { aktivitetId } = queryString.parse(location.search);
    return aktivitetId ? getFirst(aktivitetId) : undefined;
}

export const useSelectedAktivitet = (): MaybeAktivitet => {
    const dialog = useSelectedDialog();
    if (!dialog) return;
    const aktivitetData = useAktivitetContext();
    const aktivitetId = useAktivitetId() ?? dialog?.aktivitetId;
    return findAktivitet(aktivitetData, aktivitetId);
};
export const useSelectedDialog = (): DialogData | undefined => {
    const { dialoger } = useDialogContext();
    const params = useParams();
    const { dialogId } = params;
    return dialoger.find((dialog) => dialog.id === dialogId);
};
