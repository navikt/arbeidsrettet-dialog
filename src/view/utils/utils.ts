import { Aktivitet, AktivitetTypes, AlleAktivitetTypes, ArenaAktivitet } from '../../utils/aktivitetTypes';
import { pathnamePrefix } from '../../utils/UseApiBasePath';

export const erArenaAktivitet = (aktivitetId: string | null | undefined): boolean =>
    !!aktivitetId && aktivitetId.startsWith('ARENA');

export const getBasename = (fnr?: string): string => {
    return settSammenmedSlasher(pathnamePrefix, fnr);
};

export const settSammenmedSlasher = (...ss: Array<string | undefined>): string => {
    const slasherImellom = ss
        .filter((s: string | undefined): s is string => !!s)
        .flatMap<string>((s) => s.split('/'))
        .filter((s) => s.trim() !== '')
        .join('/');
    if (slasherImellom !== '') return '/' + slasherImellom;
    else return '';
};

const erGCP = (): boolean => window.location.hostname.endsWith('intern.nav.no');

export const getContextPath = (): string => (erGCP() ? '' : '/veilarbpersonflatefs');

export const getAktivitetType = (aktivitet: Aktivitet | ArenaAktivitet): AlleAktivitetTypes => {
    if (aktivitet.type === AktivitetTypes.EKSTERN_AKTIVITET) {
        return aktivitet.eksternAktivitet!!.type;
    }
    return aktivitet.type;
};

export const runningOnGithubPages = process.env.REACT_APP_USE_HASH_ROUTER === 'true';
