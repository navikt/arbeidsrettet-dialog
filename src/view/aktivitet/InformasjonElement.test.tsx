import React from 'react';
import { mount, shallow } from 'enzyme';
import InformasjonElement from './InformasjonElement';
import { EtikettLiten, Undertekst } from 'nav-frontend-typografi';
import EksternLenke from '../../felleskomponenter/EksternLenke';
import '../../utils/SetupEnzyme';

describe('<InformasjonElement />', () => {
    it('skal vise fritekst hvis det er oppgitt', () => {
        const merkelapptekst = 'merkelappteksten';
        const verdi = 'Verdi på merkelapp';
        const wrapper = mount(<InformasjonElement merkelapptekst={merkelapptekst} verdi={verdi} />);

        expect(wrapper.find(EtikettLiten).exists).toBeTruthy();
        expect(wrapper.find(EtikettLiten).text()).toEqual(merkelapptekst);

        expect(wrapper.find(Undertekst).exists()).toBeTruthy();
        expect(wrapper.find(Undertekst).text()).toEqual(verdi);
    });

    it('skal vise children om det er oppgitt', () => {
        const merkelapptekst = 'merkelappteksten';
        const jsx = (
            <InformasjonElement merkelapptekst={merkelapptekst} verdi="">
                <EksternLenke lenke="nav.no" />
            </InformasjonElement>
        );
        const wrapper = mount(jsx);

        expect(wrapper.find(EtikettLiten).exists).toBeTruthy();
        expect(wrapper.find(EtikettLiten).text()).toEqual(merkelapptekst);
        expect(wrapper.find(EksternLenke).exists()).toBeTruthy();
    });

    it('skal skjules om hverken tekst eller children er oppgitt', () => {
        const merkelapptekst = 'merkelappteksten';
        const wrapper = shallow(<InformasjonElement merkelapptekst={merkelapptekst} verdi={null} />);
        expect(wrapper.find(EtikettLiten).exists()).toBeFalsy();
    });

    it('skal matche snapshot når EksternLenke er child', () => {
        const merkelapptekst = 'merkelappteksten';
        const jsx = (
            <InformasjonElement merkelapptekst={merkelapptekst} verdi="">
                <EksternLenke lenke="nav.no" />
            </InformasjonElement>
        );
        const wrapper = mount(jsx);

        expect(wrapper).toMatchSnapshot();
    });
});
