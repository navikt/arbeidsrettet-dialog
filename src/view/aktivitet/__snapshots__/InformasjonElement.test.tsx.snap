// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InformasjonElement /> skal matche snapshot n√•r EksternLenke er child 1`] = `
<InformasjonElementRaw
  merkelapptekst="merkelappteksten"
>
  <div>
    <Undertekst>
      <TypografiBase
        type="undertekst"
      >
        <p
          className="typo-undertekst"
        >
          merkelappteksten
        </p>
      </TypografiBase>
    </Undertekst>
    <EksternLenke
      lenke="nav.no"
    >
      <Lenke
        href="http://nav.no"
      >
        <a
          className="lenke"
          href="http://nav.no"
        >
          nav.no
          <ForwardRef(SvgEksternLenke)>
            <svg>
              ekstern-lenke.svg
            </svg>
          </ForwardRef(SvgEksternLenke)>
        </a>
      </Lenke>
    </EksternLenke>
  </div>
</InformasjonElementRaw>
`;
